plugins/localfile-securitymodule/src/main/java/org/hyperledger/besu/plugin/services/securitymodule/localfile/LocalFileSecurityModulePlugin.java

```diff
@@ -0,0 +1,104 @@
+import org.hyperledger.besu.plugin.BesuContext;
+import org.hyperledger.besu.plugin.BesuPlugin;
+import org.hyperledger.besu.plugin.services.BesuConfiguration;
+import org.hyperledger.besu.plugin.services.PicoCLIOptions;
+import org.hyperledger.besu.plugin.services.SecurityModuleService;
+import org.hyperledger.besu.plugin.services.securitymodule.SecurityModule;
+import org.hyperledger.besu.plugin.services.securitymodule.localfile.configuration.LocalFileSecurityModuleCLIOptions;
+
+import java.io.File;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class LocalFileSecurityModulePlugin implements BesuPlugin {
+  public static final String PICOCLI_NAMESPACE = "localfile-security-module";
+
+  private static final Logger LOG = LogManager.getLogger();
+  static final String SECURITY_MODULE_NAME = "localfile";
+  private final boolean isDocker = Boolean.getBoolean("besu.docker");
+  private final LocalFileSecurityModuleCLIOptions cliOptions =
+      new LocalFileSecurityModuleCLIOptions();
+
+  @Override
+  public void register(final BesuContext context) {
+    LOG.debug("Registering plugin");
+    registerCliOptions(context);
+    registerSecurityModule(context);
+  }
+
+  private void registerCliOptions(final BesuContext context) {
+    if (!isFullInstantiation()) {
+      return; // don't register cli options in docker mode
+    }
+
+    final PicoCLIOptions picoCLIOptions =
+        context
+            .getService(PicoCLIOptions.class)
+            .orElseThrow(
+                () ->
+                    new IllegalStateException(
+                        "Expecting a PicoCLIOptions service to register CLI options with, but none found."));
+
+    picoCLIOptions.addPicoCLIOptions(PICOCLI_NAMESPACE, cliOptions);
+  }
+
+  private void registerSecurityModule(final BesuContext context) {
+    context
+        .getService(SecurityModuleService.class)
+        .orElseThrow(
+            () ->
+                new RuntimeException(
+                    "Security Module Service Not available, cannot register Service Module: "
+                        + SECURITY_MODULE_NAME))
+        .registerSecurityModule(SECURITY_MODULE_NAME, this::createFileBasedKeyPairSecurityModule);
+  }
+
+  private SecurityModule createFileBasedKeyPairSecurityModule(
+      final BesuConfiguration besuConfiguration) {
+    final File privateKeyFile = nodePrivateKeyFile(besuConfiguration);
+    final SECP256K1.KeyPair keyPair = KeyPairUtil.loadKeyPair(privateKeyFile);
+    return new KeyPairSecurityModule(keyPair);
+  }
+
+  private File nodePrivateKeyFile(final BesuConfiguration besuConfiguration) {
+    final File nodePrivateKeyFile = isFullInstantiation() ? cliOptions.getPrivateKeyFile() : null;
+
+    return nodePrivateKeyFile != null
+        ? nodePrivateKeyFile
+        : KeyPairUtil.getDefaultKeyFile(besuConfiguration.getDataPath());
+  }
+
+  private boolean isFullInstantiation() {
```

Comment:
I know its stolen, but its terribly named - can we change this to be more meaningful - i.e. or "containerised"? even "inDocker"?
Is this function even required - does it add value over the literal variable?

Reply:
agreed. Modified the code to use `isDocker` variable directly.